### PR TITLE
Pr nimrod@sca jira ticketing with new aggregation logic

### PR DESCRIPTION
**Pull request by NimrodGolan**
### Description

Support SCA Jira tickets aggregation

### References

https://dev.azure.com/CxFlow/CxFlow/_workitems/edit/136?src=WorkItemMention&src-action=artifact_link

### Testing
TBD

### CODE REVIEW

File: src/main/java/com/checkmarx/flow/service/JiraService.java [hunk 1]
```diff
@@ -37,13 +37,13 @@
...
-    public static final String JIRA_ISSUE_KEY = "%s%s @ %s [%s]%s";
-    public static final String JIRA_ISSUE_KEY_2 = "%s%s @ %s%s";
-    public static final String JIRA_ISSUE_KEY_3 = "%s%s Vulnerable Package @ %s [%s]%s";
-    public static final String JIRA_ISSUE_BODY = "*%s* issue exists @ *%s* in branch *%s*";
-    public static final String JIRA_ISSUE_BODY_2 = "*%s* issue exists @ *%s*";
-    public static final String JIRA_ISSUE_BODY_3 = "*%s Vulnerable Package* issue exists @ *%s* in branch *%s*";
+    private static final int MAX_RESULTS_ALLOWED = 1000000;
+    private static final String JIRA_ISSUE_KEY = "%s%s @ %s [%s]%s";
+    private static final String JIRA_ISSUE_KEY_2 = "%s%s @ %s%s";
+    private static final String JIRA_ISSUE_KEY_3 = "%s<%s> %.1f: <%s> in %s and %s @ <%s.%s>%s";
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/service/JiraService.java:43]: remove the <>

File: src/main/java/com/checkmarx/flow/service/JiraService.java [hunk 2]
```diff
@@ -292,7 +292,11 @@ private String createIssue(ScanResults.XIssue issue, ScanRequest request) throws
             if (useBranch) {
                 List<ScanResults.ScaDetails> scaDetails = issue.getScaDetails();
                 if (scaDetails != null) {
-                    summary = String.format(JIRA_ISSUE_KEY_3, issuePrefix, scaDetails.get(0).getFinding().getSeverity(), scaDetails.get(0).getVulnerabilityPackage().getName(), branch, issuePostfix);
+                    ScanResults.ScaDetails details = scaDetails.get(0);
+                    summary = String.format(JIRA_ISSUE_KEY_3, issuePrefix, details.getFinding().getSeverity(),
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/service/JiraService.java:296]: details.getFindings() is used 3 times, please extract it to a variable.

Comment (ofersk) [reply src/main/java/com/checkmarx/flow/service/JiraService.java:296]: also extract the package (used 2 time), this will improve readability

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
### Checkmarx scan completed
[Full Scan Details](http://EC2AMAZ-JP2I9QF/CxWebClient/ViewerMain.aspx?scanid=1010276&projectid=110)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|435
Low|4993
Informational|44

#### Violation Summary
Severity|Count
---|---

#### Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---

**Pull request merged**